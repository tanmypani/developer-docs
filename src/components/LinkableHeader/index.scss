@import '../../assets/styles/variables.scss';

.linkable-header {
    .link-element {
        vertical-align: text-top;
        padding-left: $padding-xs;
        font-size: $font-size-doc;
        display: none;
    }

    &:hover .link-element {
        display: inline-block;
    }
}