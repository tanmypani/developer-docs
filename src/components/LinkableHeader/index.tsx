import React, { useEffect } from 'react';
import './index.scss';

type LinkableSectionProps = {
    children: React.ReactNode;
    tag: string,
    id: string
};

const LinkableHeader = (props: LinkableSectionProps) => {
    const Tag = props.tag as keyof JSX.IntrinsicElements;
    return (
        <Tag id={props.id} className="linkable-header">
            {props.children}
            <a href={`#${props.id}`} className="link-element">ðŸ”—</a>
        </Tag>
    );
};

export default LinkableHeader;
