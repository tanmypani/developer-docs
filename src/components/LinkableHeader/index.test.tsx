import React from 'react'
import { render } from '@testing-library/react'
import LinkableHeader from './index'

describe('Linkable Header', () => {
    it('Should render the right content', () => {
        const { container } = render(<LinkableHeader tag="h4" id="test-id">Test Header Content</LinkableHeader>);
        expect(container).toMatchSnapshot();
    })
})
