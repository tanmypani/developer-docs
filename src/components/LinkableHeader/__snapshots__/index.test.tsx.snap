// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linkable Header Should render the right content 1`] = `
<div>
  <h4
    class="linkable-header"
    id="test-id"
  >
    Test Header Content
    <a
      class="link-element"
      href="#test-id"
    >
      ðŸ”—
    </a>
  </h4>
</div>
`;
